diff --git a/harness/atomicsHelper.js b/harness/atomicsHelper.js
index 135c16e..b19f2ef 100644
--- a/harness/atomicsHelper.js
+++ b/harness/atomicsHelper.js
@@ -222,10 +222,14 @@ $262.agent.waitUntil = function(typedArray, index, expected) {
  *   }
  */
 $262.agent.timeouts = {
-  yield: 100,
-  small: 200,
-  long: 1000,
-  huge: 10000,
+//  yield: 100,
+//  small: 200,
+//  long: 1000,
+//  huge: 10000,
+  yield: 20,
+  small: 20,
+  long: 100,
+  huge: 1000,
 };
 
 /**
diff --git a/harness/regExpUtils.js b/harness/regExpUtils.js
index 2abfee3..e7c07b1 100644
--- a/harness/regExpUtils.js
+++ b/harness/regExpUtils.js
@@ -5,24 +5,27 @@ description: |
     Collection of functions used to assert the correctness of RegExp objects.
 ---*/
 
+if ($262 && typeof $262.codePointRange === "function") {
+    /* use C function to build the codePointRange (much faster with
+       slow JS engines) */
+    codePointRange = $262.codePointRange;
+} else {
+    codePointRange = function codePointRange(start, end) {
+        const codePoints = [];
+        let length = 0;
+        for (codePoint = start; codePoint < end; codePoint++) {
+            codePoints[length++] = codePoint;
+        }
+        return String.fromCodePoint.apply(null, codePoints);
+    }
+}
+
 function buildString({ loneCodePoints, ranges }) {
-  const CHUNK_SIZE = 10000;
-  let result = Reflect.apply(String.fromCodePoint, null, loneCodePoints);
-  for (let i = 0; i < ranges.length; i++) {
-    const range = ranges[i];
-    const start = range[0];
-    const end = range[1];
-    const codePoints = [];
-    for (let length = 0, codePoint = start; codePoint <= end; codePoint++) {
-      codePoints[length++] = codePoint;
-      if (length === CHUNK_SIZE) {
-        result += Reflect.apply(String.fromCodePoint, null, codePoints);
-        codePoints.length = length = 0;
-      }
+    let result = String.fromCodePoint.apply(null, loneCodePoints);
+    for (const [start, end] of ranges) {
+        result += codePointRange(start, end + 1);
     }
-    result += Reflect.apply(String.fromCodePoint, null, codePoints);
-  }
-  return result;
+    return result;
 }
 
 function testPropertyEscapes(regex, string, expression) {
diff --git a/test/language/comments/hashbang/use-strict.js b/test/language/comments/hashbang/use-strict.js
index 962614c..541c115 100644
--- a/test/language/comments/hashbang/use-strict.js
+++ b/test/language/comments/hashbang/use-strict.js
@@ -10,7 +10,7 @@ description: >
 info: |
     HashbangComment::
       #! SingleLineCommentChars[opt]
-flags: [raw]
+flags: [raw, noStrict]
 features: [hashbang]
 ---*/
 
diff --git a/test/language/expressions/class/private-getter-brand-check-multiple-evaluations-of-class-realm-function-ctor.js b/test/language/expressions/class/private-getter-brand-check-multiple-evaluations-of-class-realm-function-ctor.js
index 48ed9a8..f486a2c 100644
--- a/test/language/expressions/class/private-getter-brand-check-multiple-evaluations-of-class-realm-function-ctor.js
+++ b/test/language/expressions/class/private-getter-brand-check-multiple-evaluations-of-class-realm-function-ctor.js
@@ -16,7 +16,7 @@ info: |
   PrivateBrandCheck(O, P)
     1. If O.[[PrivateBrands]] does not contain an entry e such that SameValue(e, P.[[Brand]]) is true,
       a. Throw a TypeError exception.
-features: [class, class-methods-private]
+features: [class, class-methods-private, cross-realm]
 ---*/
 
 let classStringExpression = `
diff --git a/test/language/expressions/class/private-getter-brand-check-multiple-evaluations-of-class-realm.js b/test/language/expressions/class/private-getter-brand-check-multiple-evaluations-of-class-realm.js
index 5f3b3b5..7991e7c 100644
--- a/test/language/expressions/class/private-getter-brand-check-multiple-evaluations-of-class-realm.js
+++ b/test/language/expressions/class/private-getter-brand-check-multiple-evaluations-of-class-realm.js
@@ -16,7 +16,7 @@ info: |
   PrivateBrandCheck(O, P)
     1. If O.[[PrivateBrands]] does not contain an entry e such that SameValue(e, P.[[Brand]]) is true,
       a. Throw a TypeError exception.
-features: [class, class-methods-private]
+features: [class, class-methods-private, cross-realm]
 flags: [noStrict]
 ---*/
 
diff --git a/test/language/expressions/class/private-method-brand-check-multiple-evaluations-of-class-realm-function-ctor.js b/test/language/expressions/class/private-method-brand-check-multiple-evaluations-of-class-realm-function-ctor.js
index 71c536a..514cb2f 100644
--- a/test/language/expressions/class/private-method-brand-check-multiple-evaluations-of-class-realm-function-ctor.js
+++ b/test/language/expressions/class/private-method-brand-check-multiple-evaluations-of-class-realm-function-ctor.js
@@ -16,7 +16,7 @@ info: |
   PrivateBrandCheck(O, P)
     1. If O.[[PrivateBrands]] does not contain an entry e such that SameValue(e, P.[[Brand]]) is true,
       a. Throw a TypeError exception.
-features: [class, class-methods-private]
+features: [class, class-methods-private, cross-realm]
 ---*/
 
 let classStringExpression = `
diff --git a/test/language/expressions/class/private-method-brand-check-multiple-evaluations-of-class-realm.js b/test/language/expressions/class/private-method-brand-check-multiple-evaluations-of-class-realm.js
index 929e9c0..7a887c4 100644
--- a/test/language/expressions/class/private-method-brand-check-multiple-evaluations-of-class-realm.js
+++ b/test/language/expressions/class/private-method-brand-check-multiple-evaluations-of-class-realm.js
@@ -16,7 +16,7 @@ info: |
   PrivateBrandCheck(O, P)
     1. If O.[[PrivateBrands]] does not contain an entry e such that SameValue(e, P.[[Brand]]) is true,
       a. Throw a TypeError exception.
-features: [class, class-methods-private]
+features: [class, class-methods-private, cross-realm]
 flags: [noStrict]
 ---*/
 
diff --git a/test/language/expressions/class/private-setter-brand-check-multiple-evaluations-of-class-realm-function-ctor.js b/test/language/expressions/class/private-setter-brand-check-multiple-evaluations-of-class-realm-function-ctor.js
index 1227a41..9f42680 100644
--- a/test/language/expressions/class/private-setter-brand-check-multiple-evaluations-of-class-realm-function-ctor.js
+++ b/test/language/expressions/class/private-setter-brand-check-multiple-evaluations-of-class-realm-function-ctor.js
@@ -16,7 +16,7 @@ info: |
   PrivateBrandCheck(O, P)
     1. If O.[[PrivateBrands]] does not contain an entry e such that SameValue(e, P.[[Brand]]) is true,
       a. Throw a TypeError exception.
-features: [class, class-methods-private]
+features: [class, class-methods-private, cross-realm]
 ---*/
 
 let classStringExpression = `
diff --git a/test/language/expressions/class/private-setter-brand-check-multiple-evaluations-of-class-realm.js b/test/language/expressions/class/private-setter-brand-check-multiple-evaluations-of-class-realm.js
index d73056f..6a943c9 100644
--- a/test/language/expressions/class/private-setter-brand-check-multiple-evaluations-of-class-realm.js
+++ b/test/language/expressions/class/private-setter-brand-check-multiple-evaluations-of-class-realm.js
@@ -16,7 +16,7 @@ info: |
   PrivateBrandCheck(O, P)
     1. If O.[[PrivateBrands]] does not contain an entry e such that SameValue(e, P.[[Brand]]) is true,
       a. Throw a TypeError exception.
-features: [class, class-methods-private]
+features: [class, class-methods-private, cross-realm]
 flags: [noStrict]
 ---*/
 
